sdk/servicebus/service-bus/src/serviceBusMessage.ts

```diff
@@ -983,42 +935,56 @@ export class ServiceBusMessage implements ReceivedMessage {
    *
    * @returns Promise<void>
    */
-  async deadLetter(options?: DeadLetterOptions): Promise<void> {
-    const error: AmqpError = {
-      condition: Constants.deadLetterName
-    };
-    if (options) {
-      error.info = {
-        DeadLetterReason: options.deadletterReason,
-        DeadLetterErrorDescription: options.deadLetterErrorDescription
-      };
-    }
+  async deadLetter(propertiesToModify?: DeadLetterOptions & { [key: string]: any }): Promise<void> {
     log.message(
       "[%s] Deadlettering the message with id '%s'.",
       this._context.namespace.connectionId,
       this.messageId
     );
-    if (this._context.requestResponseLockedMessages.has(this.lockToken!)) {
-      await this._context.managementClient!.updateDispositionStatus(
-        this.lockToken!,
-        DispositionStatus.suspended,
-        {
-          deadLetterReason: error.condition,
-          deadLetterDescription: error.description,
-          sessionId: this.sessionId
-        }
-      );
 
-      // Remove the message from the internal map of deferred messages
-      this._context.requestResponseLockedMessages.delete(this.lockToken!);
+    const actualPropertiesToModify: Partial<DeadLetterOptions> = {
+      ...propertiesToModify
+    };
+
+    // these two fields are handled specially and don't need to be in here.
+    delete actualPropertiesToModify.deadLetterErrorDescription;
+    delete actualPropertiesToModify.deadletterReason;
+
+    const dispositionStatusOptions: DispositionStatusOptions = {
+      propertiesToModify: actualPropertiesToModify,
+      deadLetterReason: propertiesToModify?.deadletterReason,
+      deadLetterDescription: propertiesToModify?.deadLetterErrorDescription
+    };
+    return this.settleMessage(DispositionStatus.suspended, dispositionStatusOptions);
+  }
+
+  private async settleMessage(
+    operation: DispositionStatus,
+    options?: DispositionStatusOptions
+  ): Promise<void> {
+    const dispositionType = getDispositionType(operation);
+    const isDeferredMessage = this._context.requestResponseLockedMessages.has(this.lockToken!);
```

Comment:
Sorry, I got confused by this. `isDeferredMessage` also applies to deadlettered messages too right?

Reply:
No. Deferred messages are messages on which you called `defer()`. They are different from dead lettered messages.

Reply:
Sorry, I wasn't clear. Do we need a similar check for deadlettered messages? Looking at the code that changed (https://github.com/Azure/azure-sdk-for-js/pull/8406/files#diff-5bcda9b74c142dce01964e0412155572L1001-L1010), deadlettered messages also used to perform the same check `this._context.requestResponseLockedMessages.has(this.lockToken!);`, and use the management link to settle.

Reply:
Messages can be settled in 4 ways - complete, abandon, defer, deadletter.
- `deferring` a message would put the message in a deferred queue. 
- To receive a message from deferred queue, we rely on the management link. 
- Once we receive such message from a deferred queue, its locktoken is preserved in `requestResponseLockedMessages`(A map of locked messages received using the management client, this map only contains locktokens of deferred messages).
- To settle the received message from a deferred queue, we need to rely on the management link. Which is why we perform this if-check `this._context.requestResponseLockedMessages.has(this.lockToken!);` in all the settlement methods to find out if the received message was from a deferred queue.

Reply:
> deadlettered messages also used to perform the same check this._context.requestResponseLockedMessages.has(this.lockToken!);, and use the management link to settle.

What you are referring to is dead lettering a deferred message, and not dealing with already dead lettered messages.

When we receive deferred messages, the `this._context.requestResponseLockedMessages` is updated to track them. This is because settling via a normal receiver link is not possible for deferred messages as these messages are not received via a normal receiver link in the first place.

Reply:
Thanks for all the discussion. Resolving the comment.